lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1885,7 +1899,22 @@ def _resolve_depenency_map(
         ))
         error_msg_lines.append(pre_release_hint)
         raise AnsibleError('\n'.join(error_msg_lines)) from dep_exc
+    except CollectionDependencyResolutionError as resolution_err:
+        # NOTE: This is unlikely to ever be hit as the exception is only used
+        # NOTE: within `resolvelib` as a base for `ResolutionImpossible` and
+        # NOTE: `ResolutionTooDeep`. Still, we catch it explicitly, just in
+        # NOTE: case.
+        raise AnsibleError(str(resolution_err)) from resolution_err
+    except CollectionDependencyRequirementsConflicted as req_conflict:
+        # NOTE: This is unlikely to ever be hit as the exception looks to be
+        # NOTE: either suppressed or transformed into `ResolutionImpossible`
+        # NOTE: within `resolvelib`. Still, we catch it explicitly, just in
+        # NOTE: case.
+        raise AnsibleError(str(req_conflict)) from req_conflict
     except CollectionDependencyInconsistentCandidate as dep_exc:
+        display.vvvvv(
+            f'Collection dependency inconsistent candidate: {dep_exc !s}',
+        )
```

Comment:
I'd be in favor of showing this information by default, especially since the error message doesn't give any clues that more informational will be provided with `-vvvvv`. IMO it is a bug that `find_matches` doesn't ensure all codepaths check the matches with `is_satisfied_by`, which is the only way to trigger the issue (I think...is that correct?).

Using a contrived example, here's the -vvvvv message and error:
```console
Collection dependency inconsistent candidate: Provided candidate <ns.col:1.0.0 of type 'dir' from /home/shertel/.ansible/tmp/ansible-local-1371163oxzeoi1w/tmpymmqcykj/col0kym4k42> does not satisfy <ns.col:1.0.0 of type 'dir' from /home/shertel/.ansible/tmp/ansible-local-1371163oxzeoi1w/tmpymmqcykj/col0kym4k42>, <ns.col:2.0.0 of type 'dir' from /home/shertel/.ansible/tmp/ansible-local-1371163oxzeoi1w/tmpymmqcykj/col0bvs07p5>
ERROR! Failed to resolve the requested dependencies map. A package index offered us an invalid candidate ns.col:1.0.0 (dependency of git collection from a Git repo, git collection from a Git repo) and it does not satisfy all of the following requirements:
* ns.col:1.0.0
* ns.col:2.0.0
Hint: Pre-releases hosted on Galaxy or Automation Hub are not installed by default unless a specific version is requested. To enable pre-releases globally, use --pre.
```

I think the confusing part (which you haven't modified) is `... ns.col:1.0.0 (dependency of git collection from a Git repo, git collection from a Git repo)`, since it doesn't help me find the FQCN causing the issue.

Reply:
As I understand it, this might be happening before we know what's inside the dist. Like when Galaxy API returns metadata over JSON but then we download and unpack the actual artifact and there's a mismatch in it. This is what the inconsistent candidate exception represents.

I suppose we could do
```suggestion
        display.display(
            f'Collection dependency inconsistent candidate: {dep_exc !s}',
        )
```
if you prefer. But it seems to me like a detail for troubleshooting. I've been lacking it for this purpose.

Reply:
I was thinking it could use `display.error` possibly. It's a bug if this exception is triggered at all. We can make it never happen by making sure `find_matches` uses `is_satisfied_by`. Currently it doesn't for concrete artifacts.